app/controllers/api/providers_controller.rb

```diff
@@ -91,7 +91,12 @@ def format_provider_custom_attributes(attribute)
         attribute["value"] = attribute.delete("field_type").safe_constantize.parse(attribute["value"])
       end
       attribute["section"] ||= "metadata" unless @req.action == "edit"
+      if attribute["section"].present? && !(CustomAttribute::ALLOWED_API_SECTIONS.include? attribute["section"])
+        raise ForbiddenError, "Unauthorized attribute section specified: #{attribute["section"]}"
```

Comment:
we'd still not be allowing it with the BadRequestError (400),  403's definition is at higher level than a custom attribute section of a resource.